test/spec-example/shmem_team_split_2D.c

```diff
@@ -0,0 +1,58 @@
+/*
+ *  This test program is derived from an example program in the
+ *  OpenSHMEM specification.
+ */
+
+#include <stdio.h>
+#include <shmem.h>
+#include <shmemx.h>
+#include <math.h>
+
+/* Find x and y such that x * y == npes and abs(x - y) is minimized. */
+void find_x_and_y_dims(int npes, int *x, int *y) {
+    for(int divider = ceil(sqrt(npes)); divider >= 1; divider--)
+        if (npes % divider == 0) {
+            *x = divider;
+            *y = npes / divider;
+            return;
+        }
+}
+
+int main(void)
+{
+  int xdim, ydim;
+
+  shmem_init();
+  int pe = shmem_my_pe();
+  int npes = shmem_n_pes();
+
+  find_x_and_y_dims(npes, &ydim, &xdim);
+
+  if (shmem_my_pe() == 0) printf("xdim = %d, ydim = %d\n", xdim, ydim);
+
+  int zdim = (npes / (xdim*ydim)) + ( ((npes % (xdim*ydim)) > 0) ? 1 : 0 );
```

Comment:
Also, you can simplify this expression `( ((npes % (xdim*ydim)) > 0) ? 1 : 0 )` to `(npes % (xdim*ydim) ? 1 : 0)`.

Reply:
Oh right... `zdim` is 1 no matter what `npes` is, because `find_x_and_y_dims` assures `xdim * ydim == npes`... 😞 

I'll see how nasty the cube search is... maybe this modification isn't going to work out.